driver_test.go

```diff
@@ -516,6 +516,55 @@ func TestPodmanDriver_logNomad(t *testing.T) {
 	must.StrContains(t, stdoutLog, stderrMagic)
 }
 
+// Check default plugin logger
+// TODO: Can we make this nicer?
```

Comment:
This just tests that the default is set, preferably I'd like to provide a special `PluginConfig` that explicitly sets this value?

Reply:
I think this would be fine for now. In the future it would be interesting to turn https://github.com/hashicorp/nomad-driver-podman/blob/main/api/api.go into an interface and inject a mock into the driver that doesn't actually calls Podman and allows us to inspect the config the driver submitted to `ContainerCreate`.

But that's a whole different work stream 😅 

If you do feel like adding more through testing, another scenario to test would be to ensure the task config overwrites the plugin config.

This could be done with a test similar to this, but the plugin is configured with `journald` and the task with `nomad`. `nomad` should be picked as the correct log driver.